cmdlineparserlib/include/cmd_line_parser.h

```diff
@@ -0,0 +1,48 @@
+#include<map>
+#include<unordered_map>
+#include<vector>
+#include<string> 
+#include<iostream>
+#include<algorithm>
+#include <cstdio>
+
+namespace cmd_parser
+{
+	
+	class AddOptions
+	{
+	protected:
+
+		std::vector<std::string> so;//for  short command 
+		std::vector<std::string> lo;//for long command
+		std::vector<std::string> s_desc;//for short description
+		std::vector<std::string> l_desc;//for long description
+		std::vector<std::string> data; //for data type of command 
```

Comment:
Make struct/ class out it, instead of storing individual variables.
e.g
class commad {
                std::string so;//for  short command 
		std::string lo;//for long command
		std::string s_desc;//for short description
		std::string l_desc;//for long description
		variant data; //for data type of command ( need to use std::variant / create your own variant as discussed earler)
}

and create a map of Commond;
std::map<string,command> commanlist;